lesson04.md

```diff
@@ -0,0 +1,464 @@
+```
+
+Is the same as:
+
+```js
+let myAge = 42;
+myAge += 1;   // same as myAge = myAge + 1
+```
+
+---
+
+### Quiz: What's the price?
+
+```js
+let price = 10;
+// there's a 50 percent sale!
+price /= 2;
+```
+
+Result: `5` <!-- .element: class="fragment" -->
+
+---
+
+### Operator ++ and --
+
+* Operator `++` and `--` increase (or decrease) the value of a variable by 1
+
+```js
+let myAge = 42;
+myAge++; // same as myAge = myAge + 1
+```
+
+---
+
+### Quiz: What's the age?
+
+```js
+let myAge = 42;
+myAge--;
+```
+
+Result: `41` <!-- .element: class="fragment" -->
+
+---
+
+## Recap: Operator chaining
+
+---
+
+### Operator chaining
+
+We can "chain" operators, for example:
+
+```js
+1 + 2 + 3
+(1 + 2) + 3  // first, 1 + 2 is computed (left to right)
+(3) + 3      // returns 6
+```
+
+---
+
+### Operator chaining
+
+* Is the following code valid?
+
+```js
+5 < 6 < 7
+```
+
+* Answer: It's valid JavaScript, but pointless  <!-- .element: class="fragment" -->
+
+---
+
+### Operator chaining (cont.)
+
+* Let's break it down:
+
+```js
+5 < 6 < 7
+```
+
+```js
+(5 < 6) < 7   // 5 < 6 is true
+```
+<!-- .element: class="fragment" -->
+
+```js
+true < 7      // pointless comparison!
+```
+<!-- .element: class="fragment" -->
+
+* comparing a boolean to a number makes no sense. Don't chain comparison operators.
+<!-- .element: class="fragment" -->
+
+---
+
+### Quiz
+
+* What does the following return?
+
+```js
+5 === 5 === 5
+```
+
+```js
+(5 === 5) === 5
+true === 5         // returns false
+```
+<!-- .element: class="fragment" -->
+
+---
+
+## Operator precedence
+
+---
+
+### Operator precedence
+
+What do you think is the result?
+
+```js
+2 + 2 * 2
+```
+
+Result: `6` <!-- .element: class="fragment" -->
+
+---
+
+### Operator precedence (cont.)
+
+* JavaScript supports precedence (priority) for operators
+* Operators with higher precedence are evaluated before operators with lower precedence
+* Full List: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Operator_Precedence
+* Multiplication (15) has higher precedence than addition (14) - multiplication wins
+
+---
+
+### Operator precedence (cont.)
+
+* Parentheses have the highest precedence (21) and always win:
+
+```js
+(2 + 2) * 2   // 8
+2 + (2 * 2)   // 6
+```
+
+* When in doubt, use parentheses!
+
+---
+
+### Quiz
+
+What is `x` ?
+
+```js
+let x = 4 + 6 < 3 + 8;
+```
+
+Result: `true` <!-- .element: class="fragment" -->
+
+---
+
+### Best practice: Use variables instead of chaining
+
+Which of the two is more readable?
+
+```js
+let x = 4 + 6 < 3 + 8;
+```
+
+```js
+let sum1 = 4 + 6;
+let sum2 = 3 + 6;
+let x = sum1 < sum2;
+```
+
+---
+
+## Logical Operators
+
+---
+
+### Logical Operators
+
+These operators work on `boolean` values:
+
+* `&&` AND
+* `||` OR
+* `! ` NOT
+
+---
+
+### Logical NOT !
+
+Logical NOT negates a boolean:
+
+```javascript
+!true  // false
+!false // true
+```
+
+---
+
+Example: Carlo doesn't like people who like pizza hawaii.
```

Comment:
lol, that's a bit harsh though 🤣 

Maybe just something with `isPizzaHawaiiGood = true;`

`"Carlo likes pizza hawaii:", !isPizzaHawaiiGood`

Reply:
Haha ok, I changed it to:

Example: Carlo's Pizzeria doesn't serve pizza hawaii.

```js
let likesPizzaHawaii = true;
console.log("Would like to go to Carlo's Pizzeria:", !likesPizzaHawaii);
```

Reply:
my own pizzeria! A dream come true!